recipes/full_finetune_distributed.py

```diff
@@ -357,10 +357,16 @@ def _setup_data(
         iterable datasets and streaming datasets are not supported.
         """
         world_size, rank = utils.get_world_size_and_rank()
-        ds = config.instantiate(
-            cfg_dataset,
-            tokenizer=self._tokenizer,
-        )
+
+        if isinstance(cfg_dataset, ListConfig):
+            datasets = [
+                config.instantiate(single_cfg_dataset, tokenizer=self._tokenizer)
+                for single_cfg_dataset in cfg_dataset
+            ]
```

Comment:
Not necessary for this PR, but it'd be nice to have a utility like 

```
def instantiate_list_config(cfg: ListConfig, **kwargs) -> List[Any]:
    return [
        instantiate(element, kwargs) for element in cfg
    ]
```

to reduce copypasta across the recipes a bit. cc @RdoubleA to tackle in a follow-up